guides/payments.md

```diff
@@ -0,0 +1,220 @@
+with it, then Solidus will attempt to process the payment. Otherwise, the
+payment will need to be processed manually.
+
+If the `PaymentMethod` object is configured to auto-capture payments, then the
+`Payment#purchase!` method will be called, which will call
+`PaymentMethod#purchase` like this:
+
+```ruby
+payment_method.purchase(<amount>, <source>, <gateway options>)
+```
+
+If the payment is *not* configured to auto-capture payments, the
+`Payment#authorize!` method will be called, with the same arguments as the
+`purchase` method above:
+
+```ruby
+payment_method.authorize(<amount>, <source>, <gateway options>)
+```
+
+How the payment is actually put through depends on the `PaymentMethod`
+sub-class' implementation of the `purchase` and `authorize` methods.
+
+The returned object from both the `purchase` and `authorize` methods on the 
+payment method objects must be an `ActiveMerchant::Billing::Response` object.
+This response object is then stored (in YAML) in the `spree_log_entries` table.
+Log entries can be retrieved with a call to the `log_entries` association on any
+`Payment` object.
+
+If the `purchase!` route is taken and is successful, the payment is marked as
+`completed`. If it fails, it is marked as `failed`. If the `authorize` method is
+successful, the payment is transitioned to the `pending` state so that it can be
+manually captured later by calling the `capture!` method. If it is unsuccessful,
+it is also transitioned to the `failed` state.
+
+***
+Once a payment has been saved, it also updates the order. This may trigger the
+`payment_state` to change, which would reflect the current payment state of the
+order. The possible states are:
+
+* `balance_due`: Indicates that payment is required for this order
+* `failed`: Indicates that the last payment for the order failed
+* `credit_owed`: This order has been paid for in excess of its total
+* `paid`: This order has been paid for in full.
+***
+
+!!!
+You may want to keep tabs on the number of orders with a `payment_state` of
+`failed`. A sudden increase in the number of such orders could indicate a
+problem with your credit card gateway and most likely indicates a serious
+problem affecting customer satisfaction. You should check the latest
+`log_entries` for the most recent payments in the store if this is happening.
+!!!
+
+### Log Entries
+
+Responses from payment gateways within Solidus are typically
+`ActiveMerchant::Billing::Response` objects. When Solidus handles a response
+from a payment gateway, it will serialize the object as YAML and store it in the
+database as a log entry for a payment. These responses can be useful for
+debugging why a payment has failed.
+
+You can get a list of these log entries by calling the `log_entries` on any 
+`Spree::Payment` object. To get the `Active::Merchant::Billing::Response` out of these `Spree::LogEntry` objects, call the `details` method.
+
+## Supported Gateways
```

Comment:
I'm not sure if we should maintain a list of supported gateways here. I think that reference to another resource where we can find gateway extensions could be better.

What do you think?

I was thinking in [extensions.solidus.io](http://extensions.solidus.io). I like how show the builds for each version, so you can know if a project is maintained, but don't have a way to filter the extensions. Another option could be [solition](http://soliton.nebulab.it) that find extensions from github (finding by repository description and name, I think so). Another option?

Reply:
I agree we'll probably don't need to keep this list, even because the direction on gateways is trying to build a gem for each geteway for several reason. They don't share any logic in `solidus_gateway` gem and 99% of times stores just use one gateway from that list.